creativecloud/scripts/utils.js

```diff
@@ -38,3 +38,49 @@ export const [setLibs, getLibs] = (() => {
     }, () => libs,
   ];
 })();
+
+function getDecorateAreaFn() {
+  let lcpImgSet = false;
+
+  // Load LCP image immediately
+  const eagerLoad = (lcpImg) => {
+    lcpImg?.setAttribute('loading', 'eager');
+    lcpImg?.setAttribute('fetchpriority', 'high');
+    if (lcpImg) lcpImgSet = true;
+  };
+
+  async function loadLCPImage(area = document, { fragmentLink = null } = {}) {
+    const firstBlock = area.querySelector('body > main > div > div');
+    let fgDivs = null;
+    switch (true) {
+      case firstBlock?.classList.contains('changebg'): {
+        firstBlock.querySelector(':scope > div:nth-child(1)').querySelectorAll('img').forEach(eagerLoad);
+        const { getConfig } = await import(`${getLibs()}/utils/utils.js`);
+        import(`${getConfig().codeRoot}/creativecloud/deps/interactive-marquee-changebg/changeBgMarquee.js`);
+        break;
+      }
+      case firstBlock?.classList.contains('marquee'):
+        firstBlock.querySelectorAll('img').forEach(eagerLoad);
+        break;
+      case firstBlock?.classList.contains('interactive-marquee'):
+        firstBlock.querySelector(':scope > div:nth-child(1)').querySelectorAll('img').forEach(eagerLoad);
+        fgDivs = firstBlock.querySelector(':scope > div:nth-child(2)').querySelectorAll('div:not(:first-child)');
+        fgDivs.forEach((d) => eagerLoad(d.querySelector('img')));
+        break;
+      case !!fragmentLink:
+        if (window.document.querySelector('a.fragment') === fragmentLink && !window.document.querySelector('img[loading="eager"]')) {
+          eagerLoad(area.querySelector('img'));
+        }
+        break;
+      default:
+        if (!fragmentLink) eagerLoad(area.querySelector('img'));
+        break;
+    }
+  }
+
+  return (area, options) => {
+    if (!lcpImgSet) loadLCPImage(area, options);
```

Comment:
isn't lcpImgSet always false?

Reply:
`false` is the initial value and will be set to `true` once the `eagerLoad` function is called successfully. This flag ensures that only one fragment in a list of fragments will be eagerly loaded which is the LCP one.